src/imgui/ImGuiDebugger.cc

```diff
@@ -620,21 +621,30 @@ void ImGuiDebugger::drawSlots(MSXCPUInterface& cpuInterface, Debugger& debugger)
 				if (ImGui::TableNextColumn()) { // segment
 					auto* device = cpuInterface.getVisibleMSXDevice(page);
 					Debuggable* romBlocks = nullptr;
+					MSXRom *rom = nullptr;
 					if (auto* mapper = dynamic_cast<MSXMemoryMapperBase*>(device)) {
 						ImGui::StrCat(mapper->getSelectedSegment(page));
-					} else if (!dynamic_cast<RomPlain*>(device) &&
+					} else if ((rom = dynamic_cast<MSXRom*>(device)) &&
 						(romBlocks = debugger.findDebuggable(device->getName() + " romblocks"))) {
-						// TODO we should query the actual bankSize instead this heuristic
+						auto type = rom->getRomType();
+						unsigned blockSize = RomInfo::getBlockSize(type);
+						unsigned numBlocks = blockSize ? 0x4000 / blockSize : 0;
 						std::array<uint8_t, 4> segments;
-						for (auto sub : xrange(4)) {
-							segments[sub] = romBlocks->read(addr + 0x1000 * sub);
+						for (auto sub : xrange(numBlocks)) {
+							segments[sub] = romBlocks->read(addr + blockSize * sub);
 						}
-						if ((segments[0] != segments[1]) || (segments[2] != segments[3])) { // 4kB
-							ImGui::StrCat('R', segments[0], '/', segments[1], '/', segments[2], '/', segments[3]);
-						} else if (segments[0] != segments[2]) { // 8kB
-							ImGui::StrCat('R', segments[0], '/', segments[2]);
-						} else { // 16kB
+						switch (numBlocks) {
+						case 1: // 16kB
 							ImGui::StrCat('R', segments[0]);
+							break;
+						case 2: // 8kB
+							ImGui::StrCat('R', segments[0], '/', segments[1]);
+							break;
+						case 4: // 4kB
+							ImGui::StrCat('R', segments[0], '/', segments[1], '/', segments[2], '/', segments[3]);
+							break;
+						default:
+							ImGui::TextUnformatted("-"sv);
 						}
 					} else {
 						ImGui::TextUnformatted("-"sv);
```

Comment:
Looks good. (If I have time) I'll probably merge this PR this evening.

But may I suggest one change. Replace the code inside the `if ((rom = dynamic_cast...` with
It's both shorter and more general (handles all block sizes).
Caveat: I have NOT tested this.

    if (unsigned blockSize = RomInfo::getBlockSize(rom->getRomType())) {
        std::string text;
        char separator = 'R';
        for (int offset = 0; offset < 0x4000; offset += blockSize) {
            strAppend(text, separator, romBlocks->read(addr + offset));
            separator = '/';
        }
        ImGui::TextUnformatted(text);
    } else {
        ImGui::TextUnformatted("-"sv);
    }

Reply:
I've tested it. It's OK.